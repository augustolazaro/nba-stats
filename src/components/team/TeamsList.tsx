import * as React from 'react'
import graphql from 'graphql-tag'
import styled from 'styled-components'

import TeamLogo from '../common/TeamLogo'

import createQueryRenderer from '../../hocs/createQueryRenderer'

const Container = styled.div`
  display: flex;
  flex-direction: column;
`

const Content = styled.div`
  display: flex;
  flex-wrap: wrap;
`

const TeamCard = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 20px;
  background-color: ${props => props.color};
  cursor: pointer;

  @media (max-width: 700px) {
    flex: 1;
  }
`

const Header = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 20px;
`

const PageTitle = styled.h1``


type Props = {
  query: {
    teams: Array<{}>,
  },
  history: any,
}

const TeamsList = ({ query, history }: Props) => {
  const goToDetails = (id: string) => {
    history.push(`/team/${id}`)
  }

  return (
    <Container>
      <Header>
        <PageTitle>Teams</PageTitle>
      </Header>

      <Content>
        {query.teams.map((team: any, index: number) => (
          <TeamCard key={index} color={team.colors[0]} onClick={() => goToDetails(team.id)}>
            <TeamLogo src={team.logo} size={200} />
          </TeamCard>
        ))}
      </Content>
    </Container>
  )
}

export default createQueryRenderer(TeamsList, {
  query: graphql`
    query TeamsListQuery {
      teams {
        id
        name
        logo
        colors
      }
    }
  `,
})