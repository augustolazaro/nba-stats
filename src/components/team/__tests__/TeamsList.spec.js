import * as React from 'react'
import { MockedProvider } from 'react-apollo/test-utils';
import graphql from 'graphql-tag';
import * as renderer from 'react-test-renderer';
import wait from 'waait'

import TeamsList from '../TeamsList'
import { ThemeProvider } from 'styled-components';
import { theme } from '../../../index'

const mock = [
  {
    request: {
      query: graphql`
        query TeamsListQuery {
          teams {
            id
            name
            logo
            colors
          }
        }
      `
    },
    result: {
      data: {
        teams: [
          { id: '1', name: 'Chicago Bulls', logo: '', colors: ['#000'] },
          { id: '2', name: 'Cleveland Cavaliers', logo: '', colors: ['#000'] },
          { id: '3', name: 'Brooklin Nets', logo: '', colors: ['#000'] },
        ],
      }
    },
  }
]

it('should render teams list with no errors', async () => {
  const component = renderer.create(
    <MockedProvider mocks={mock} addTypename={false}>
      <ThemeProvider theme={theme}>
        <TeamsList />
      </ThemeProvider>
    </MockedProvider>
  );

  await wait(0)

  expect(component.toJSON()).toMatchSnapshot()
})