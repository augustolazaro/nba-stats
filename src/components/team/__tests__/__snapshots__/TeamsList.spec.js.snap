// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render teams list with no errors 1`] = `
<div
  className="sc-bxivhb glCqXO"
>
  <div
    className="sc-bZQynM cSOULE"
  >
    <h1
      className="sc-htoDjs bZzhDu"
    >
      Teams
    </h1>
  </div>
  <div
    className="sc-ifAKCX cZFrkc"
  >
    <div
      className="sc-EHOje jCEXIt"
      color="#000"
      onClick={[Function]}
    >
      <div
        className="sc-gzVnrw kHfKKv"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height={25}
          stroke="rgba(255, 2555, 255, .3)"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={25}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
          />
        </svg>
      </div>
      <img
        className="sc-bdVaJa hVnfYN"
        height={200}
        src=""
        width={200}
      />
    </div>
    <div
      className="sc-EHOje jCEXIt"
      color="#000"
      onClick={[Function]}
    >
      <div
        className="sc-gzVnrw kHfKKv"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height={25}
          stroke="rgba(255, 2555, 255, .3)"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={25}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
          />
        </svg>
      </div>
      <img
        className="sc-bdVaJa hVnfYN"
        height={200}
        src=""
        width={200}
      />
    </div>
    <div
      className="sc-EHOje jCEXIt"
      color="#000"
      onClick={[Function]}
    >
      <div
        className="sc-gzVnrw kHfKKv"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height={25}
          stroke="rgba(255, 2555, 255, .3)"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={25}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
          />
        </svg>
      </div>
      <img
        className="sc-bdVaJa hVnfYN"
        height={200}
        src=""
        width={200}
      />
    </div>
  </div>
</div>
`;
